### ISSUE TITLE
GitLab Activity Feed: Double space line break in issue comment breaks sync

### ISSUE THREAD

**Issue by sbondorf** [ViennaRSS/vienna-rss#1545]
**Describe the bug**
GitLab has an activity overview (Project Information > Activity) RSS feed that includes, among other news items, comments to issues. The comments are written in markdown, like comments here on GitHub. To add a line break  
(like this one),  
double space is used. Yet, this breaks Vienna synchronization. Other RSS readers (tested with NetNewsWire) keep sync'ing, though.

**To Reproduce**
- Subscribe to a GitLab repository's activity feed
- Create an issue
- Comment on the issue, include a double space to break a line

**Screenshots**
I've set up this public repository to demonstrate the issue: https://gitlab.com/sbondorf/rsstestsvienna2/-/issues/1
![Screen Shot 2022-02-14 at 15 56 40](https://user-images.githubusercontent.com/5295332/153887745-552f09c3-6a3e-458c-84a2-e2f86563c184.png)
<img width="437" alt="Screen Shot 2022-02-14 at 15 57 00" src="https://user-images.githubusercontent.com/5295332/153887801-9a33cd6b-8cbc-4f3f-acb6-5842c9126f95.png">

**Please complete the following information:**
 - Version 3.8.0 Beta 2
- macOS 12.2.1, Intel

**Additional information:**
Was present in previous versions, too, sorry for not reporting earlier

**Comment by barijaona** [comment]
Gitlab newbie question : what is the address of the RSS feed ?

**Comment by sbondorf** [comment]
https://gitlab.com/sbondorf/rsstestsvienna2/-/issues.atom

![Screen Shot 2022-05-02 at 07 44 15](https://user-images.githubusercontent.com/5295332/166189769-5e539504-ed08-4b5f-8b97-86a0fcfdcd3c.png)

**Comment by barijaona** [comment]
I think you are referring to https://gitlab.com/sbondorf/rsstestsvienna2.atom ?

**Comment by sbondorf** [comment]
You are right! 

I posted the Atom feed for issues only only. I took it "by accident", didn't check my issue description and I didn't know it would work around the issue. 

The feed you posted matches the Issue description as it is the feed of the entire activity view containing all project changes (Wiki, members, etc.). That does not work with Vienna.

The sample project only has a single issue, it breaks Vienna's ability to subscribe to the Activity Feed but not the Issues Feed. Interesting find, I hope that helps debugging.

**Comment by barijaona** [comment]
This issue as well as issue #1544 are caused by a mix of XHTML and HTML tags…

Inside of a `<summary type="xhtml">...</summary>`construct, GitLab inserts some `<img… >` or `<br>` tags, but the correct syntax should be respectively `<img... />` and `<br />` (XHTML versions).

If I reverse commit 97480f3, I am able to parse the above mentioned GitLab feeds, but alas, this would reintroduce the #1073 crash…

I will try playing with other options of NSXMLDocument to work around this.

**Comment by sbondorf** [comment]
Thank you for getting to the bottom of this!

Maybe it is worthwhile to upstream the issue by filing a bug with GitLab, providing them with all the details?

I checked the open issues [here](https://gitlab.com/gitlab-org/gitlab/-/issues), filtered by combinations of "feed", "atom", "html", "xhtml" but that didn't turn up a known (and open) issue.

**Comment by barijaona** [comment]
Yes, you can report the issue to GitLab and suggest inserting the HTML into CDATA instead of claiming it's XHTML. 

However, I will try to make Vienna able to handle malformed feeds like this.

**Comment by barijaona** [comment]
I notice that you have spotted https://gitlab.com/gitlab-org/gitlab/-/issues/345797 in the meantime

**Comment by sbondorf** [comment]
> Yes, you can report the issue to GitLab and suggest inserting the HTML into CDATA instead of claiming it's XHTML

Is that sufficient as bug information. Will do then.

> I notice that you have spotted https://gitlab.com/gitlab-org/gitlab/-/issues/345797 in the meantime

Exactly, that would be nice, too, but only if it doesn't make more problems of the current kind ...

**Comment by barijaona** [comment]
You can comment on the current GitLab issue, providing https://gitlab.com/sbondorf/rsstestsvienna.atom and https://gitlab.com/sbondorf/rsstestsvienna2.atom as supplemental examples and copying there my explanations. 

My suggestion to use CDATA marking is probably the easiest solution, but I don't know if and how the upcoming GitLab 15 will change the situation.

**Comment by sbondorf** [comment]
By current GitLab issue, do you refer to https://gitlab.com/gitlab-org/gitlab/-/issues/345797 ?

I fail to see the connection, would have opened a new issue

**Comment by barijaona** [comment]
I have added a comment to the current GitLab issue : https://gitlab.com/gitlab-org/gitlab/-/issues/345797#note_936145356

**Comment by barijaona** [comment]
@josh64x2 Do you have any suggestion regarding this issue ?

I think we reach the limits of NSXMLDocument 's "closed box" approach. Catching errors and recovering from them would not be easy and would require parsing NSError's localizedDescription.

I am tempted to replace most of RichXMLParser with [RSParser](https://github.com/Ranchero-Software/RSParser), a NetNewsWire framework. The package claims to require macOS 10.15, but my guess is it can easily be back ported to 10.12.

It would also be a source of synergies between Mac open source projects… 😄
